### PR TITLE
Restored WorkloadIdentityFederation support

### PR DESCRIPTION
**Pull request by starkmsu**
**Task name**: AzureFunctionAppV1, AzureFunctionAppV2

**Description**: Restored WorkloadIdentityFederation support

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

**Review by FinVamp1 — COMMENTED** [review]
Hello,

While in general I understand using Azure-Arm-Rest-v2 helps avoid duplication of code it does mean that we have lost the targeted logging that differentiates Functions from App Service. I note that you have removed the Functions flavour of the App Service and Kudu classes which means that we lose the targeted logging for Functions,. console.log("##vso[telemetry.publish area=TaskDeploymentMethod;feature=AzureFunctionAppDeployment]" + authMethodtelemetry); . This may seem like a nit but we are trying to distinguish between the tasks so we can look into the telemetry and dientify failures before the task gets to us? Perhaps we can pass in parameter to the common logic?

File: Tasks/AzureFunctionAppV2/operations/FileTransformsUtility.ts
[the remark this answers is not in the document]
Comment (starkmsu) [reply Tasks/AzureFunctionAppV2/operations/FileTransformsUtility.ts:1]: Kudu service is used in AzureRmWebAppDeploymentProvider, added metrics logging for "AzureFunctionAppDeployment" based on the last change in the common library

File: Tasks/AzureFunctionAppV1/task.loc.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
   "author": "Microsoft Corporation",
   "version": {
     "Major": 1,
-    "Minor": 221,
-    "Patch": 105
+    "Minor": 222,
```
Comment (ozanovus) [inline Tasks/AzureFunctionAppV1/task.loc.json:20]: Versions should be 223 as we passed cut off.

File: Tasks/AzureFunctionAppV1/azure-arm-rest/azure-arm-app-service.ts [hunk 2]
```diff
@@ -1,824 +0,0 @@
...
-            }
-
-            return response.body;
-        }
-        catch(error) {
-            throw Error(`Failed to get Virtual Network Connections. Error: ${this._client.getFormattedError(error)}`);
-        }
-    }
-
-    public async getSitePrivateEndpointConnections(): Promise<any> {
```
[the remark this answers is not in the document]
Comment (starkmsu) [reply Tasks/AzureFunctionAppV1/azure-arm-rest/azure-arm-app-service.ts:774]: It is already in arm-rest-v2 common npm package - https://github.com/microsoft/azure-pipelines-tasks/blob/da0575664ca1e113200c1124154b93d54b7b8a6c/common-npm-packages/azure-arm-rest-v2/azure-arm-app-service.d.ts#L45

File: Tasks/AzureFunctionAppV1/azure-arm-rest/azure-arm-app-service-kudu.ts [hunk 3]
```diff
@@ -1,710 +0,0 @@
...
-
-            return error.message;
-        }
-        if(error && error.statusCode) {
-            return `${error.statusMessage} (CODE: ${error.statusCode})`;
-        }
-        return error;
-    }
-    
-    public async validateZipDeploy(webPackage: string, queryParameters?: Array<string>): Promise<any> {
```
[the remark this answers is not in the document]
Comment (starkmsu) [reply Tasks/AzureFunctionAppV1/azure-arm-rest/azure-arm-app-service-kudu.ts:667]: Currently this method is in azure-arm-app-service-kudu - https://github.com/microsoft/azure-pipelines-tasks/blob/da0575664ca1e113200c1124154b93d54b7b8a6c/common-npm-packages/azure-arm-rest-v2/azure-arm-app-service-kudu.d.ts#L32

**Pull request merged**